### PR TITLE
[inductor] Added decomposition for upsample_nearest_exact Nd

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Added decomposition for upsample_nearest_exact: 1d, 2d, 3d

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -2412,26 +2433,50 @@ def upsample_nearest3d_vec(input, output_size, scale_factors):
...
+        # index_f32 = (output_index + 0.5) * scale - 0.5
+        # input_index = round(index_f32)
+        # Same as Pillow and Scikit-Image/Scipy ndi.zoom
         osize = output_size[d]
-        output_indices = torch.arange(osize, dtype=input_dtype, device=input.device)
         isize = input.shape[-num_spatial_dims + d]
         scale = isize / (isize * scales[d]) if scales[d] is not None else isize / osize
-        input_indices = (output_indices * scale).to(torch.int64)
+
+        output_indices = torch.arange(osize, dtype=input_dtype, device=input.device)
```
Comment (vadimkantorov) [inline torch/_decomp/decompositions.py:2478]: I hope this doesn't break for torch.float16 or other small-bitwidth dtypes (given large osize).

So maybe input_dtype here should always be torch.float32, even if input dtype is float16

Comment (vfdev-5) [reply torch/_decomp/decompositions.py:2478]: good point, in aten upsample nearest we are using `using opmath_t = at::opmath_type<scalar_t>;` which translates to : 
```
uint8 -> float  (as we specifically define `input_scalar_type = at::ScalarType::Float;` for bytes)
half -> float
BFloat16 -> float
double -> double
```

**Comment by janeyx99** [comment]
@voznesenskym added you as reviewer due to the inductor/lowering, but please reassign if there is a better reviewer

**Review by lezcano — APPROVED** [review]

**Pull request closed without merging**